### PR TITLE
Print with nuxt + pagedjs

### PR DESCRIPTION
**Pull request by usu**
Merges Nuxt infrastructure into devel. Integrates work done by @carlobeltrame in PR https://github.com/ecamp/ecamp3/pull/2418. Focus is on infrastructure, not on print content.

Printing on deployment doesn't work, because necessary changes to [`continuous-deployment.yml`](https://github.com/ecamp/ecamp3/pull/2480/files#diff-af36bb0ba98217223305c14372a4e893c25a61dcfa63a78b32eb3f37c6cd1285) are not yet in devel. Manual deployment tested and available under https://ecamp3.mybytes.ch/

### What it can/should do
- Printing in local dev environment via locally running browserless docker image
- Printing on deployment via browserless cloud (currenty, my token is registered in our deploy secrets)
- Removes all old legacy code (weasy & puppeteer workers) and test code
- Picasso in landscape layout can be generated from Picasso view (Print icon in top right corner)

### What is not implemented
Needs to be covered in separate PRs/issues:
- Cleanup of print content (currently still a bit a mess between real content & demo content and some duplicate code)
- PDF generation of routes other than /picasso
- Managing both nuxt and react print methods in the frontend in parallel. My suggestion would be to utilize [vue-feature-toggle](https://github.com/bassdman/vue-feature-toggle) to enable/disable both print methods and let the user choose which method to use if both are enabled. This would allow us to decide at a late stage or potentially even ship both methods and react quickly if one doesn't scale well.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -47,8 +47,7 @@ services:
       # default port for Xdebug 3 is 9003
       # idekey=VSCODE if you are debugging with VSCode
       XDEBUG_CONFIG: >-
-        client_host=docker-host
-        idekey=PHPSTORM
+        client_host=docker-host idekey=PHPSTORM
```
Comment (BacLuc) [inline docker-compose.yml:50]: is this necessary?

File: docker-compose.yml [hunk 2]
```diff
@@ -222,6 +184,13 @@ services:
...
+    build:
+      context: ./print/browserless
+    container_name: 'ecamp3-browserless'
+    ports:
+      - '3010:3000'
+
 volumes:
-  db-data:
-  db-data-postgres:
+  db-data: null
```
Comment (BacLuc) [inline docker-compose.yml:195]: is this necessary?

Comment (BacLuc) [inline docker-compose.yml:195]: is this necessary?

File: frontend/src/App.vue [hunk 3]
```diff
@@ -34,15 +34,15 @@ export default {
       return this.$auth.user()
     },
     deploymentTime () {
-      const timestamp = window.environment.DEPLOYMENT_TIME
+      const timestamp = window.environment?.DEPLOYMENT_TIME
```
Comment (BacLuc) [inline frontend/src/App.vue]: should the environment be nullable?

File: frontend/src/plugins/store/index.js [hunk 4]
```diff
@@ -16,7 +16,7 @@ class StorePlugin {
     })
 
     axios.defaults.withCredentials = true
-    axios.defaults.baseURL = window.environment.API_ROOT_URL
+    axios.defaults.baseURL = window.environment?.API_ROOT_URL
```
Comment (BacLuc) [inline frontend/src/plugins/store/index.js]: should the environemnt be nullable?

File: frontend/src/router.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ const NavigationCamp = () => import(/* webpackChunkName: "navigationCamp" */ './
 /* istanbul ignore next */
 export default new Router({
   mode: 'history',
-  base: window.environment.BASE_URL || '/',
+  base: window.environment?.BASE_URL || '/',
```
Comment (BacLuc) [inline frontend/src/router.js]: should the environment be nullable?

File: frontend/src/views/camp/CampProgram.vue [hunk 6]
```diff
@@ -49,6 +60,10 @@ import ScheduleEntries from '@/components/scheduleEntry/ScheduleEntries.vue'
 import PeriodSwitcher from '@/components/camp/PeriodSwitcher.vue'
 import LocalPdfDownloadButton from '../../components/print/LocalPdfDownloadButton.vue'
 
+import axios from 'axios'
+
+const PRINT_SERVER = window.environment?.PRINT_SERVER
```
Comment (BacLuc) [inline frontend/src/views/camp/CampProgram.vue]: should the environment be nullable?

File: print/nuxt.config.js [hunk 7]
```diff
@@ -73,47 +78,30 @@ export default {
...
-    baseURL: process.env.INTERNAL_API_ROOT_URL || 'http://backend:3001/api',
+    baseURL: process.env.INTERNAL_API_ROOT_URL || 'http://caddy:3001/',
+    credentials: true,
   },
   /*
    ** Sentry module configuration
    ** See https://sentry.nuxtjs.org/sentry/options
    */
   sentry: {
     // Use a dummy DSN so that the sentry module doesn't disable itself during build
```
Comment (BacLuc) [inline print/nuxt.config.js]: can the comment be deleted?

**Review by BacLuc — APPROVED** [review]
some smaller comments

File: docker-compose.yml (hunk 1 above)
Comment (usu) [reply docker-compose.yml:50]: Strong opinion from VSCode. If it annoys, I can try to disable the formatter

File: docker-compose.yml (hunk 2 above)
Comment (usu) [reply docker-compose.yml:195]: Strong opinion from VSCode. If it annoys, I can try to disable the formatter

Comment (usu) [reply docker-compose.yml:195]: Strong opinion from VSCode. If it annoys, I can try to disable the formatter

File: print/nuxt.config.js (hunk 7 above)
Comment (usu) [reply print/nuxt.config.js]: Yes, the comment is obsolete, will remove.
I didn't check though if sentry still works. Added this to #2481

File: frontend/src/App.vue (hunk 3 above)
Comment (usu) [reply frontend/src/App.vue]: Leftover from my deployment to netlify before we had kubernetes deployment. Reverted with https://github.com/ecamp/ecamp3/pull/2480/commits/ef356d5a0c1005d7fd1341aacedb888856f59bea

File: print/package.json [hunk 8]
```diff
@@ -15,17 +15,21 @@
     "test": "jest"
   },
   "engines": {
-    "npm": "~8.2.0"
+    "npm": "~8.1.0"
```
Comment (carlobeltrame) [inline print/package.json:18]: Ist das ein Merge-Fehler?

**Review by carlobeltrame — APPROVED** [review]

File: print/package.json (hunk 8 above)
Comment (usu) [reply print/package.json:18]: Ne. Der print-Container läuft aktuell auf Node 16.13.2 welches wiederum npm 8.1.2 enthält. 

Dito übrigens beim frontend-Container. Wir haben einfach die Fehlermeldung dazu konsequent ignoriert.

**Pull request merged**